src/dispdrivers/vgacontroller.cpp

```diff
@@ -151,33 +151,95 @@ void IRAM_ATTR VGAController::VSyncInterrupt(void * arg)
 }
 
 
+std::function<uint8_t(RGB888 const &)> VGAController::getPixelLamda(PaintMode mode)
+{
+  switch (mode) {
+    case PaintMode::XOR:
+      return [&] (RGB888 const & color) { return preparePixel(color) & 63; };
+    case PaintMode::ANDNOT:
+    case PaintMode::ORNOT:
+      return [&] (RGB888 const & color) { return (~preparePixel(color) & 63) || m_HVSync; };
```

Comment:
this method returns a lambda function for the pixel bit pattern used in drawing operations

XOR operations need to have their sync bits removed, as XORing those will clear them

for the two operations that combine in a NOT of the target colour, we invert the target here, rather than in the routines that set pixels

those two NOT variants also must ensure that the sync bits are properly set, as otherwise they will have been inverted